src/open_inwoner/openklant/services.py

```diff
@@ -1200,47 +1208,94 @@ def get_or_create_partij_for_user(self, user: User) -> tuple[Partij | None, bool
 
         return partij, created
 
+    def _retrieve_digitale_addressen(
+        self,
+        subject_type: Literal["partij", "betrokkene"],
+        uuid: str,
+    ) -> list[DigitaalAdres]:
+        if subject_type == "partij":
+            paginated_data = self.client.digitaal_adres.list(
+                params={"verstrektDoorPartij__uuid": uuid}
+            )
+        else:
+            paginated_data = self.client.digitaal_adres.list(
+                params={"verstrektDoorBetrokkene__uuid": uuid}
+            )
+
+        digitale_adressen = list(pagination_helper(self.client, paginated_data))
+
+        return digitale_adressen
+
     def retrieve_digitale_addressen_for_partij(
         self, partij_uuid: str
     ) -> list[DigitaalAdres]:
-        expand_partij = self.client.partij.retrieve(
-            partij_uuid, params={"expand": ["digitaleAdressen"]}
-        )
+        return self._retrieve_digitale_addressen("partij", uuid=partij_uuid)
 
-        if (expand := expand_partij.get("_expand")) and (
-            digitale_adressen := expand.get("digitaleAdressen")
-        ):
-            return digitale_adressen
-
-        # TODO: A missing _expand can mean there are no addresses.
-        # See: https://github.com/maykinmedia/open-klant/issues/243
-        return []
+    def retrieve_digitale_addressen_for_betrokkene(
+        self, betrokkene_uuid: str
+    ) -> list[DigitaalAdres]:
+        return self._retrieve_digitale_addressen("betrokkene", uuid=betrokkene_uuid)
 
-    def filter_digitale_addressen_for_partij(
+    def _filter_digitale_addressen(
         self,
-        partij_uuid: str,
+        subject_type: Literal["partij", "betrokkene"],
+        uuid: str,
         *,
         soort_digital_adres: str,
         adressen: Iterable[DigitaalAdres] | None = None,
     ) -> list[DigitaalAdres]:
         if adressen is None:
-            adressen = self.retrieve_digitale_addressen_for_partij(partij_uuid)
+            if subject_type == "partij":
+                adressen = self.retrieve_digitale_addressen_for_partij(uuid)
+            else:  # betrokkene
+                adressen = self.retrieve_digitale_addressen_for_betrokkene(uuid)
 
         return [
             digitaal_adres
             for digitaal_adres in adressen
             if digitaal_adres["soortDigitaalAdres"] == soort_digital_adres
         ]
 
-    def get_or_create_digitaal_adres(
+    def filter_digitale_addressen_for_partij(
         self,
         partij_uuid: str,
+        *,
+        soort_digital_adres: str,
```

Comment:
```suggestion
        soort_digitaal_adres: str,
```

Would a literal make sense here?

Reply:
And if so, perhaps we can also do the `match`/`assert_never` pattern below.

Reply:
I've added `Literal`, but I'm not sure about the `match` pattern  here. There's nothing to do in any of the cases, we would just `pass` for the successful ones and `assert_never` for the fall-through...